### PR TITLE
Document recursive property conversions

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Documents the complex recursive property conversion code at a high level to make it more visible.

**Special notes for your reviewer**:

Does it need another example or three? Is that example clear enough?

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains documentation

### CODE REVIEW

**Review by babbageclunk — APPROVED** [review]
Really like the worked example here! This explanation seems really clear to me (although obviously it's hard to judge since I already have a reasonable understanding of the code).

File: docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+date: 2021-02-09
+---
+
+# Property Conversions 
+
+## Context
+
+To facilitate the use of dedicated storage variants for persistence of our custom resources, we need to codegen conversion routines that will copy all properties defined on one version of a resource to another version.
+
+Given the way resources evolve from version to version, these conversions need to handle a wide range of conversions between types that are similar, but not identical.
```
Comment (matthchr) [inline docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md:12]: very minor: Wording: "these conversions need to handle a wide range of conversions between..." seems phrase strangely?

"these conversion functions need to handle..." or conversion routines or "these conversions need to handle a wide range of transformations..."?

File: docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+A **primitive type can become optional** in a later version when a suitable default is provided, or when an alternative becomes available.
```
Comment (matthchr) [inline docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md:14]: Is this an exhaustive list of conversions? I feel like there's a bunch of cases regarding non-primitive types that are missed here? Maybe highlight that this is just the list of primitive type conversions and that the complex conversions are built out of these based on some rules, or something?

**Review by matthchr — APPROVED** [review]

File: docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md (hunk 1 above)
Comment (theunrepentantgeek) [reply docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md]: Reworded.

File: docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md (hunk 2 above)
Comment (theunrepentantgeek) [reply docs/hugo/content/architecture-decision-records/ADR-2021-02-Property-Conversions.md:16]: Not exhaustive; illustrative. Have reworded as suggested.

**Review by theunrepentantgeek — COMMENTED** [review]
Feedback addressed.

**Pull request merged**